internal/filter/filterottl/filter.go

```diff
@@ -137,41 +142,51 @@ func conditionsToStatements(conditions []string) []string {
 	return statements
 }
 
-func StandardSpanFuncs() map[string]interface{} {
+func StandardSpanFuncs() map[string]ottl.Factory[ottlspan.TransformContext] {
 	return standardFuncs[ottlspan.TransformContext]()
 }
 
-func StandardSpanEventFuncs() map[string]interface{} {
+func StandardSpanEventFuncs() map[string]ottl.Factory[ottlspanevent.TransformContext] {
 	return standardFuncs[ottlspanevent.TransformContext]()
 }
 
-func StandardMetricFuncs() map[string]interface{} {
+func StandardMetricFuncs() map[string]ottl.Factory[ottlmetric.TransformContext] {
 	return standardFuncs[ottlmetric.TransformContext]()
 }
 
-func StandardDataPointFuncs() map[string]interface{} {
+func StandardDataPointFuncs() map[string]ottl.Factory[ottldatapoint.TransformContext] {
 	return standardFuncs[ottldatapoint.TransformContext]()
 }
 
-func StandardLogFuncs() map[string]interface{} {
+func StandardLogFuncs() map[string]ottl.Factory[ottllog.TransformContext] {
 	return standardFuncs[ottllog.TransformContext]()
 }
 
-func standardFuncs[K any]() map[string]interface{} {
-	return map[string]interface{}{
-		"TraceID":     ottlfuncs.TraceID[K],
-		"SpanID":      ottlfuncs.SpanID[K],
-		"IsMatch":     ottlfuncs.IsMatch[K],
-		"Concat":      ottlfuncs.Concat[K],
-		"Split":       ottlfuncs.Split[K],
-		"Int":         ottlfuncs.Int[K],
-		"ConvertCase": ottlfuncs.ConvertCase[K],
-		"Substring":   ottlfuncs.Substring[K],
-		"drop":        drop[K],
-	}
+func standardFuncs[K any]() map[string]ottl.Factory[K] {
+	return ottl.CreateFactoryMap[K](
+		ottlfuncs.TraceIDFactory[K]{},
+		ottlfuncs.SpanIDFactory[K]{},
+		ottlfuncs.IsMatchFactory[K]{},
+		ottlfuncs.ConcatFactory[K]{},
+		ottlfuncs.SplitFactory[K]{},
+		ottlfuncs.IntFactory[K]{},
+		ottlfuncs.ConvertCaseFactory[K]{},
+		ottlfuncs.SubstringFactory[K]{},
+		dropFactory[K]{},
+	)
+}
+
+type dropFactory[K any] struct{}
+
+func (f dropFactory[K]) Name() string { return "drop" }
+
+func (f dropFactory[K]) CreateDefaultArguments() ottl.Arguments { return &struct{}{} }
```

Comment:
If it is possible to remove the pointer `&` so that the size allocated for the nop is zero.

Reply:
My other comment also applies here. I've updated this to use `nil`, which I believe should avoid the pointer allocation.